src/main/clojure/clara/rules/engine/impl.cljc

```diff
@@ -0,0 +1,45 @@
+(ns clara.rules.engine.impl)
```

Comment:
It looks like this is all protocol definitions. A convention I've seen elsewhere is to simply name this namespace "clara.rules.engine.protocols", making it clear that's what it is. Putting protocols into their own namespace is a good pattern since it makes interactive reloading easier, so I like that these are separated out.